Fix Object read and object_store tests

Jarema opened the pull request.

Jarema commented:
IMHO we have to reduce the time `object_random` takes to run because of having 1/2 byte buffer test cases with noticeable amount of bytes to read.

stevelr commented on src/object_store.rs:
```diff
@@ -281,34 +281,26 @@ impl Object {
 
-        if self.skip_next {
+        if !self.skip_next {
             self.skip_next = false;
```
If line 291 executes, `self.skip_next` is already false. Either there's a logic bug, or line 291 can be removed.

(What is the purpose of skip_next)?

Jarema replied:
Agree, line 291 can be removed.

Bug was having condition on true, instead of false. Because of that, it never actually executed.

The purpose of next is to:

subsribe to object_store subject and get messages containing data. Each loop check if there are more pending messages for the subject. If yes - set skip_next to false so next read iteration fetches next messages. If no and we're done - set it to true. Than each `read` iteration will only move data to buffer from remaining bytes.

I guess variable name can be changed to something describing actual behaviour.

Jarema replied:
@stevelr does it look clear to you know?

Jarema commented:
Issues fixed:
* potential panic of buffer
* opposite boolean check for skip_next (which resulted in never actually checking for messages in subscription)
* improved comments and docs
* fixed tests (they worked because filling vec with random bytes did nothing on 0 len vec).
* other problem around handling buffer
* renamed bool name to more descriptive one.

derekcollison reviewed (approved):
LGTM

caspervonb reviewed (commented):
LGTM

Jarema commented:
@stevelr is the author of your commit intentional (Not the one tied to your the Github Account) or just a mistake to be fixed?
